File: python/models/like.py
import dataclasses
import inspect
import typing

from google.cloud import firestore


@dataclasses.dataclass(kw_only=True, frozen=True, repr=False)
class LikeModel:
    whoLiked: typing.Optional[firestore.DocumentReference] = None
    likedProfile: typing.Optional[firestore.DocumentReference] = None

    reference: typing.Optional[firestore.DocumentReference] = None

    def to_map(self):
        data = dataclasses.asdict(self)
        del data["reference"]

        return data

    @classmethod
    def from_dict(cls, data):
        return cls(
            **{
                key: value
                for key, value in data.items()
                if key in inspect.signature(cls).parameters
            }
        )

File: models/like.ts
import type { DocumentData, DocumentReference, Timestamp } from 'firebase/firestore'

export interface ILike {
  whoLiked: DocumentReference | null
  likedProfile: DocumentReference | null
}

export class LikeModel implements ILike {
  whoLiked: DocumentReference | null
  likedProfile: DocumentReference | null

  reference: DocumentReference | null

  constructor(data: ILike, reference: DocumentReference | null) {
    this.whoLiked = data.whoLiked || null
    this.likedProfile = data.likedProfile || null

    this.reference = reference || null
  }

  toMap() {
    return {
      whoLiked: this.whoLiked,
      likedProfile: this.likedProfile,
    }
  }
}

export function mapLike(like: DocumentData) {
  return new LikeModel(
    like.data(),
    like.ref,
  )
}
